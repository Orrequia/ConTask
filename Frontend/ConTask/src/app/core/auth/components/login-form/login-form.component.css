:host {
    width: 100%;
}

form {
    display: flex;
    flex-direction: column;
    width: 100%;
}
mat-card-title,
mat-card-content {
    display: flex;
    justify-content: center;
}
mat-form-field {
    width: 100%;
    margin-bottom: 8px;
}
.loginError {
    padding: 16px;
    width: 100%;
    color: white;
    background-color: red;
}
.loginButtons {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
}

/*LO MIOOmat-card {
    max-width: 400px;
    margin: 2em auto;
    text-align: center;
}
.signin-content {
    padding: 60px 1rem;
}
.full-width-input {
    width: 100%;
}*/