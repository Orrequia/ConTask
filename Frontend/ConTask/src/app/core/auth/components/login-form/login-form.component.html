<mat-card>
  <mat-card-title>
    Login
  </mat-card-title>
  <mat-card-content>
    <form [formGroup]="loginForm" (ngSubmit)="submit()">
      <mat-form-field>
        <input matInput type="text" formControlName="alias" placeholder="Username">
        <mat-error *ngIf="loginForm.hasError('required', ['alias'])">El alias es necesario</mat-error>
      </mat-form-field>
      <mat-form-field>
        <input matInput type="password" formControlName="contrasena" placeholder="Password">
        <mat-error *ngIf="loginForm.hasError('required', ['contrasena'])">La contraseña es necesaria</mat-error>
      </mat-form-field>

      <div *ngIf="error" class="loginError">
        {{ error.error.error }}
      </div>
      <div class="loginButtons">
        <button type="submit" mat-button>Login</button>
      </div>
    </form>
  </mat-card-content>
</mat-card>

<!--EL MIOOOOOO<div class="signin-content">
  <mat-card>
    <mat-card-title>
      Login ConTask
    </mat-card-title>
    <mat-card-content>
      <form id="login-form" class="login-form" #lForm="ngForm"
            [formGroup]="form" (ngSubmit)="submitLogin()"  novalidate>
        <mat-input-container class="full-width-input" style="width: 100%; margin-bottom: 10px;"
                             [ngClass]="{'mat-input-invalid': submitted && error?.code == 1}">

          <input matInput type="text" formControlName="alias" placeholder="Alias"/>
          <mat-hint style="color: red;"
                    *ngIf="submitted && error?.code == 1 && !form.controls.alias.hasError('required')">
            {{error.message}}
          </mat-hint>
          <mat-hint style="color: red;"
                    *ngIf="submitted && form.controls.alias.hasError('required')">
            Field is mandatory
          </mat-hint>
        </mat-input-container>
        <mat-input-container class="full-width-input"
                             [ngClass]="{'mat-input-invalid': submitted && error?.code == 2}">
          <input matInput type="password" formControlName="contrasena" placeholder="Contraseña"/>
          <mat-hint style="color: red;"
                    *ngIf="submitted && error?.code == 2 && !form.controls.contrasena.hasError('required')">
            {{error.message}}
          </mat-hint>
          <mat-hint style="color: red;"
                    *ngIf="submitted && form.controls.contrasena.hasError('required')">
            Field is mandatory
          </mat-hint>
        </mat-input-container>
      </form>
    </mat-card-content>
    <mat-card-actions style="text-align: right;">
      <button mat-raised-button
              type="submit"
              form="login-form">
        Sign In
      </button>
    </mat-card-actions>
  </mat-card>
</div>

<!--<h1>Electron App with Angular Material by Systemjs</h1>

<br>
with angular material <button mat-raised-button color="primary">Click Me !!</button>
<br>

And datepicker
<br>

<mat-input-container>
  <input matInput [matDatepicker]="startPicker" placeholder="Start {{column}}">
  <mat-datepicker-toggle matSuffix [for]="startPicker"></mat-datepicker-toggle>
  <mat-datepicker #startPicker></mat-datepicker>
</mat-input-container>

<br>

<mat-form-field class="example-form-field">
  <input matInput type="text" placeholder="Clearable input" [(ngModel)]="value"/>
  <button mat-button *ngIf="value" matSuffix mat-icon-button aria-label="Clear" (click)="value=''">
    <mat-icon>close</mat-icon>
  </button>
</mat-form-field>

<mat-toolbar color="primary">
  Angular Material 2 App
</mat-toolbar>

<div class="basic-container">

  <mat-form-field>
    <mat-select placeholder="Sample" [disabled]="selectDisabled">
      <mat-option *ngFor="let opt of [1, 2, 3, 4]" [value]="opt">
        Option {{ opt }}
      </mat-option>
    </mat-select>
  </mat-form-field>

  <button (click)="selectDisabled = false">Enable</button>
  <button (click)="selectDisabled = true">Disable</button>

  <pre>disabled: {{ selectDisabled }}</pre>

</div>-->