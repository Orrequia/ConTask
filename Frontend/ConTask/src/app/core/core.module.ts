import {NgModule, Optional, SkipSelf} from '@angular/core';
import { AuthGuardService } from './auth/services/auth-guard.service';

@NgModule({
    declarations: [  ],
    imports: [],
    providers: [
        AuthGuardService
    ],
    bootstrap: []
})
export class CoreModule {
    constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
        if (parentModule) {
            throw new Error(
                'CoreModule is already loaded. Import it in the AppModule only');
        }
    }
}
