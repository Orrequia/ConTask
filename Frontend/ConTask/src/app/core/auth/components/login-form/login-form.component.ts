import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import {Validators, FormGroup, FormControl } from '@angular/forms';
import { Authenticate } from '../../models/authenticate.model';

@Component({
    selector: 'app-login-form',
    templateUrl: './login-form.component.html',
    styleUrls: ['./login-form.component.css']
})
export class LoginFormComponent implements OnInit {
    @Input() error: string | null;

    @Input() set disabled(isDisabled: boolean) {
        if (isDisabled) {
            this.loginForm.disable();
        } else {
            this.loginForm.enable();
        }
    }

    @Output() submitted = new EventEmitter<Authenticate>();

    loginForm = new FormGroup({
        alias: new FormControl('', [Validators.required]),
        contrasena: new FormControl('', [Validators.required])
    });

    ngOnInit() {}

    submit() {
        const value: Authenticate = this.loginForm.value;

        if (this.loginForm.valid) {
            this.submitted.emit(value);
        }
    }
}


/*LO MIOO    public form: FormGroup;
    public submitted: Boolean = false;
    public error: Observable<string>;
    public loading: Observable<boolean>;
    private alive = true;

    constructor(private formBuilder: FormBuilder,
                private store: Store<AppState>) {
    }

    public ngOnInit() {
        this.form = this.formBuilder.group({
            alias: ['', Validators.required],
            contrasena: ['', Validators.required],
        });

        this.error = this.store.select(getAuthenticationError);
        this.loading = this.store.select(isAuthenticationLoading);

        this.store.select(isAuthenticated)
            .takeWhile(() => this.alive)
            .filter(authenticated => authenticated)
            .subscribe(value => this.store.dispatch(
                new RouterActions.Go({
                    path: ['/empresa']})));
    }

    public ngOnDestroy() {
        this.alive = false;
    }

    public submit(): void {

        if (this.form.valid) {
            const loginObject = new LoginObject(this.form.value);

            this.store.dispatch(new AuthenticateAction(loginObject));
        }

        /*if (this.loginForm.valid) {
            this.authenticationService.login(new LoginObject(this.loginForm.value)).subscribe(
                data => this.correctLogin(data),
                error => this.error = JSON.parse(error._body)
            );
        }
        this.submitted = true;
    }
    /*private correctLogin(data: Sesion) {
        this.storageService.setCurrentSesion(data);
        this.router.navigate(['/main/empresa']);
    }
}*/

