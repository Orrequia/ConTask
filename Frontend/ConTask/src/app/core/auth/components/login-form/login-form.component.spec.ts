import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { createComponentFixture } from 'angular-testing-library';
import { LoginFormComponent } from './login-form.component';

describe('Login Form Component', () => {
    const fixture = createComponentFixture({
        component: LoginFormComponent
    });

    beforeEach(async() => {
        await fixture.compile();
    });

    it('should compile', () => {
        expect(fixture).toMatchSnapshot();
    });
});

/*LO MIOimport { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { LoginComponent } from './login-form.component';

describe('LoginComponent', () => {
  let component: LoginComponent;
  let fixture: ComponentFixture<LoginComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ LoginComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
*/