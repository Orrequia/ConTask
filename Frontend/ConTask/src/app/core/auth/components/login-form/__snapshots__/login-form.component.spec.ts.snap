// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Component should compile 1`] = `
<app-login-form
  loginForm={[Function FormGroup]}
  submitted={[Function EventEmitter]}
>
  <mat-card>
    <mat-card-title>
       Login 
    </mat-card-title>
    <mat-card-content>
      <form>
        <mat-form-field>
          <input
            formcontrolname="alias"
            matinput=""
            placeholder="Username"
            type="text"
          />
          
          <mat-error>
            El alias es necesario
          </mat-error>
        </mat-form-field>
        <mat-form-field>
          <input
            formcontrolname="contrasena"
            matinput=""
            placeholder="Password"
            type="password"
          />
          
          <mat-error>
            La contrase√±a es necesaria
          </mat-error>
        </mat-form-field>
        
        <div
          class="loginButtons"
        >
          <button
            mat-button=""
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </mat-card-content>
  </mat-card>
</app-login-form>
`;
